### PR TITLE
Add Notmuch lexer

### PR DESCRIPTION
**Pull request by dan-santana**

### CODE REVIEW

File: pygments/lexers/textfmts.py [hunk 1]
```diff
@@ -295,3 +297,86 @@ class TodotxtLexer(RegexLexer):
...
+        'part': [
+            (r'\f(?:part|attachment)}\n', Keyword, '#pop'),
+            (r'\f(?:part|attachment){\s*', Keyword, ('#push', 'part-attr')),
+            (r'^Non-text part: .*\n', Comment),
+            (r'(?s)(.*?(?=\f(?:part|attachment)}\n))', _highlight_code),
+        ],
+    }
+
+    def analyse_text(text):
+        return 1.0 if text.startswith('\fmessage{') else 0.0
```
Comment (Anteru) [inline pygments/lexers/textfmts.py]: That seems overly confident? Probably better to return `0.01` in that case?

Comment (birkenfeld) [reply pygments/lexers/textfmts.py]: If that's really a formfeed character, confidence is probably pretty well justified :smile:

Comment (Anteru) [reply pygments/lexers/textfmts.py]: Oh, missed the `\f`. I guess that's solved then, we just need a sample file (starting with a formfeed.)

**Comment by dan-santana** [comment]
> Thanks for the new lexer! Can you please provide some simple example file so we can ensure it works?

Ooops, I forgot to commit the example file! it's there now :)

File: pygments/lexers/textfmts.py [hunk 2]
```diff
@@ -295,3 +297,86 @@ class TodotxtLexer(RegexLexer):
             (r'\s+', IncompleteTaskText),
         ],
     }
+
+
+class NotmuchLexer(RegexLexer):
+    """
+    For `Notmuch <https://notmuchmail.org/>`_ email text format.
+
+    .. versionadded:: 2.4
```
Comment (birkenfeld) [inline pygments/lexers/textfmts.py]: it's 2.5 now :)

File: pygments/lexers/textfmts.py [hunk 3]
```diff
@@ -295,3 +297,86 @@ class TodotxtLexer(RegexLexer):
...
+
+        try:
+            if self.body_lexer:
+                lexer = get_lexer_by_name(self.body_lexer)
+            else:
+                lexer = guess_lexer(code.strip())
+        except ClassNotFound:
+            lexer = get_lexer_by_name('text')
+
+        for item in do_insertions([], lexer.get_tokens_unprocessed(code)):
```
Comment (birkenfeld) [inline pygments/lexers/textfmts.py]: I don't think `do_insertions` with an empty insertions list does anything other than a `for ... in lexer.get_tokens_...`.

Comment (dan-santana) [reply pygments/lexers/textfmts.py]: fixed

File: pygments/lexers/textfmts.py (hunk 2 above)
Comment (dan-santana) [reply pygments/lexers/textfmts.py]: fixed!

**Pull request merged**